src/examples/wasm/Makefile

```diff
@@ -0,0 +1,15 @@
+export: clean wasm_exec
```

Comment:
Including a clean target as a dependency of a regular target does not work when you use parallel tasks (`-j4` for example). I would recommend removing this and perhaps including `rm -rf ./html` here.

What you can also do, is not remove everything each time but only create new files, which IMHO is cleaner:

```
export: wasm_exec
	tinygo build -o ./html/wasm.wasm -target wasm ./export/wasm.go
	cp ./export/wasm.js ./html/
	cp ./export/index.html ./html/

main: wasm_exec
	tinygo build -o ./html/wasm.wasm -target wasm ./main/main.go
	cp ./main/index.html ./html/

wasm_exec:
	@mkdir -p ./html
	cp ../../../targets/wasm_exec.js ./html/

clean:
	rm -rf ./html
```

By doing `mkdir -p` each time you also ensure that the html directory exists.

Reply:
Oh, reading this a second time, it's probably better to remove the html directory every time. So, something like this:

```make
export: wasm_exec
	rm -rf ./html
	mkdir -p ./html
	cp ../../../targets/wasm_exec.js ./html/
	tinygo build -o ./html/wasm.wasm -target wasm ./export/wasm.go
	cp ./export/wasm.js ./html/
	cp ./export/index.html ./html/

main: wasm_exec
	rm -rf ./html
	mkdir -p ./html
	cp ../../../targets/wasm_exec.js ./html/
	tinygo build -o ./html/wasm.wasm -target wasm ./main/main.go
	cp ./main/index.html ./html/

clean:
	rm -rf ./html
```

The duplication is rather unfortunate, however. Not sure what the best thing is, perhaps leave it as is (even though it is not compatible with `-j2` etc.).

Reply:
I'm not well versed enough in Makefiles to know, but can we disable parallel execution in the makefile somehow? I'm not sure how big of a concern it really is, if someone was trying to run it in parallel, it'd fail and I think they'd know why it failed. The current solution is simple, even if it may not cover all cases. Do you think it's okay to leave as-is?

Reply:
https://www.gnu.org/software/make/manual/make.html#index-_002eNOTPARALLEL

But on second thought maybe this is not such a big deal. The error will likely not be obvious but I suspect most people won't run with concurrency anyway, and it can easily be fixed.